Changed MixedRealityRaycastHit to use collider transform instead of RaycastHit.transform

thalbern opened the pull request:
changed MixedRealityRaycastHit to use collider transform instead of Unity.RaycastHit.transform because this one potentially returns the transform of the gameobject that has a rigidbody attached instead of the actually collision transform.

This was the cause for not being able to interact with any handles of bounding box whenever a rigidbody component was attached. 

Unitys documentation on RaycastHit.transform is not clear on what's happening. It only states that: "The Transform of the rigidbody or collider that was hit."

While digging around I found other ppl running into this as well:
https://answers.unity.com/questions/190482/raycasthittranslatecollider.html

My observation while debugging confirms what's mentioned in that thread. Whenever we've no rigidbody component attached the transform will be set to the hit colliders transform. Whenever there's a rigidbody component somewhere in hierarchy the transform will be populated with the rigidbody gameobject transform even though the hit result collider will point to a gameobject lower in the hierarchy.

## Changes
- Fixes: #4901 
- Removed obsolete comment I found in BoundingBoxHelper

## Verification
tested modifying bounding box with and without rigid body component attached
run tests

thalbern commented:
/azp run

lukastoenneMS commented on Assets/MixedRealityToolkit/Definitions/InputSystem/MixedRealityRaycastHit.cs:
```diff
@@ -35,7 +35,7 @@ public MixedRealityRaycastHit(bool raycastValid, RaycastHit hitInfo)
                 textureCoord = hitInfo.textureCoord;
                 textureCoord2 = hitInfo.textureCoord2;
-                transform = hitInfo.transform;
+                transform = hitInfo.collider.transform;
```
Maybe add a comment here to explain why it is necessary to use .collider.transform instead.

thalbern commented:
/azp run

keveleigh commented:
This feels potentially related to https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5617

ritijain commented on Assets/MixedRealityToolkit/Definitions/InputSystem/MixedRealityRaycastHit.cs:
```diff
@@ -35,7 +35,7 @@ public MixedRealityRaycastHit(bool raycastValid, RaycastHit hitInfo)
                 textureCoord = hitInfo.textureCoord;
                 textureCoord2 = hitInfo.textureCoord2;
-                transform = hitInfo.transform;
+                transform = hitInfo.collider.transform; // we're using collider transform because hitInfo.transform might point us to a different gameobject in case there's a rigidbody in the hierarchy
```
>transform = hitInfo.collider.transform; //  [](start = 16, length = 43)

This was something that was desired by one of the app teams in mrw a while back, and we had added a profile configuration property to address this - focusIndividualCompoundCollider so that apps could choose which transform they prefer to pick - 
Check out commit - [https://github.com/microsoft/MixedRealityToolkit-Unity/commit/68994887db97a081c0483c2a79db4503fe5b570b#diff-7b0c1b5a3f72bb7ada11d8478aa6fbbb](https://github.com/microsoft/MixedRealityToolkit-Unity/commit/68994887db97a081c0483c2a79db4503fe5b570b#diff-7b0c1b5a3f72bb7ada11d8478aa6fbbb)

This change feels redundant to that commit.

thalbern replied:
I wasn't aware of that. Thanks @ritijain!

thalbern commented:
@keveleigh , @ritijain is there a reason why the default value for focusIndividualCompoundCollider is false? I feel like this is something ppl can easily run into with bounding box and I would like to enable this per default.

thalbern commented:
I removed my previous changes and replaced them instead with an inspector change for bounding box that will notify users if their setup doesn't work 

![image](https://user-images.githubusercontent.com/36998103/65336977-32db9a80-dbbf-11e9-8bcb-12d6d7e88cf3.png)

ritijain commented:
> @keveleigh , @ritijain is there a reason why the default value for focusIndividualCompoundCollider is false? I feel like this is something ppl can easily run into with bounding box and I would like to enable this per default.

This was set to false by default so as not to make any functional change in default MRTK functionality. If there is enough of a case here to set it to true, i am not opposed to it.
